velox/exec/RowNumber.cpp

```diff
@@ -182,4 +349,135 @@ int64_t RowNumber::numRows(char* partition) {
 void RowNumber::setNumRows(char* partition, int64_t numRows) {
   *reinterpret_cast<int64_t*>(partition + numRowsOffset_) = numRows;
 }
+
+void RowNumber::reclaim(
+    uint64_t /*targetBytes*/,
+    memory::MemoryReclaimer::Stats& /*stats*/) {
+  if (table_ == nullptr || table_->numDistinct() == 0) {
```

Comment:
Can we add to check if spiller has been triggered on this operator, then skip reclaim from this operator and nonReclaimableSection_ flag check?

Also could you add record spilling stats at the end of this operator execution (see recordSpillStats example in the other operator), and check the spilling stats in the test through runtime stats and operator stats? Thanks!